feat(Booklore) add Booklore

albertromkes opened the pull request:
**Description**

Add Booklore

⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code
- [ ] 📜 Documentation Changes

**🧪 How Has This Been Tested?**

Tested with app-template

**📃 Notes:**
The init-container is needed for creating the nginx run container and setting the correct permissions

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [x] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made changes to the documentation
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning
- [x] I made sure the title starts with `feat(chart-name):`, `fix(chart-name):`, `chore(chart-name):`, `docs(chart-name):` or `fix(docs):`

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

albertromkes committed:
Add booklore

albertromkes committed:
correct naming for icon

albertromkes committed:
fix wrong name in chart.yaml

alfi0812 commented on charts/incubator/booklore/README.md:
```diff
@@ -0,0 +1,50 @@
+---
+title: README
+---
+
```
Autogenerated. Please remove everything except the header

alfi0812 commented on charts/incubator/booklore/values.yaml:
```diff
@@ -0,0 +1,90 @@
+  tag: v1.8.1@sha256:6b2250cd1a8e13f4e458b5dd2ffe30ed9696482061f27954a97463ffcce677a2
+securityContext:
+  container:
+    readOnlyRootFilesystem: false
```
It's prefered to run as nonRoot if possible

albertromkes replied:
mv: inter-device move failed: '/tmp/nginx.conf.tmp' to '/etc/nginx/nginx.conf'; unable to remove target: Read-only file system
su-exec: setgroups(0): Operation not permitted

Not possible I'm afraid.

alfi0812 replied:
Read only can be disabled without setting root aswell

albertromkes replied:
Tested it, but root is necessary.

alfi0812 commented on charts/incubator/booklore/values.yaml:
```diff
@@ -0,0 +1,90 @@
+                name: '{{ printf "%s-%s" .Release.Name "mariadbcreds" }}'
+                key: jdbc-mariadb
+            DATABASE_USERNAME: booklore
+            DATABASE_PASSWORD: PLACEHOLDERPASSWORD
```
Dont hardcode. Reference the mariadb values

alfi0812 commented on charts/incubator/booklore/values.yaml:
```diff
@@ -0,0 +1,90 @@
+  enabled: true
+  mariadbUsername: booklore
+  #  Example of how to set up volsync backups for the database
+  #  persistence:
```
Remove this. Not related to the chart

alfi0812 commented on charts/incubator/booklore/values.yaml:
```diff
@@ -0,0 +1,90 @@
+    enabled: true
+    mountPath: /app/data
+  #  Your own books directory (if needed)
+  #  books:
```
Wrongly Formated and should be as a doc not in the values

alfi0812 commented on charts/incubator/booklore/values.yaml:
```diff
@@ -0,0 +1,90 @@
+                key: jdbc-mariadb
+            DATABASE_USERNAME: booklore
+            DATABASE_PASSWORD: PLACEHOLDERPASSWORD
+            BOOKLORE_PORT: 6060
```
Dont use targetport if you can specify a port. Just set this to .Values... so it fetches from above

alfi0812 commented on charts/incubator/booklore/Chart.yaml:
```diff
@@ -0,0 +1,44 @@
+apiVersion: v2
+appVersion: 0.34.0
+dependencies:
+  - name: common
```
Missing mariadb dependency. See other charts

albertromkes committed:
Correct Readme. No hardcoded values. Added docs

alfi0812 commented on charts/incubator/booklore/docs/installation_notes.md:
````diff
@@ -0,0 +1,44 @@
+
+```yaml
+mariadb:
+  enabled: true
````
Have you tried to restore this before?

albertromkes replied:
Yes, multiple times. It works (on my machine).

alfi0812 replied:
If tested and worked. More something for general docs and not for chart specific.

alfi0812 replied:
Then lets move this to a General Guide on the website and not hide it in a specific chart

alfi0812 commented on charts/incubator/booklore/docs/installation_notes.md:
```diff
@@ -0,0 +1,44 @@
+     enabled: true
+     mountPath: /books
+     path: /mnt/tank/booklore/books
+     server: ${NFS_SERVER_IP}
```
Maybe generalize those path and server values.

alfi0812 commented on charts/incubator/booklore/values.yaml:
```diff
@@ -0,0 +1,64 @@
+            - |
+              set -e
+              mkdir -p /run/nginx
+              chown -R 0:0 /run/nginx
```
Let me try something there when im at my pc. This might not be needed

alfi0812 committed:
Update values.yaml

Signed-off-by: Alfred Göppel <43101280+alfi0812@users.noreply.github.com>

alfi0812 committed:
Update values.yaml

Signed-off-by: Alfred Göppel <43101280+alfi0812@users.noreply.github.com>

alfi0812 committed:
Update values.yaml

Signed-off-by: Alfred Göppel <43101280+alfi0812@users.noreply.github.com>

alfi0812 committed:
Update values.yaml

Signed-off-by: Alfred Göppel <43101280+alfi0812@users.noreply.github.com>

alfi0812 committed:
Update values.yaml

Signed-off-by: Alfred Göppel <43101280+alfi0812@users.noreply.github.com>

alfi0812 committed:
Modify installation notes for Booklore

Updated the title and removed the Mariadb backup section from the installation notes.

Signed-off-by: Alfred Göppel <43101280+alfi0812@users.noreply.github.com>
